### PR TITLE
✨Autocomplete suggestion task selected

### LINKED ISSUE #254: Update the StartEditReducer's AutocompleteSuggestionTapped action to handle Task suggestions
## Description
When you select a task suggestion, it sets the project of the editable time entry and the task of the editable time entry as well.
If the project is in a workspace different than the one of the time entry, the time entry workspace is changed as well
Selecting a project suggestion should also remove the word that starts with an @ from the description (do so by checking the cursor position as well)

## Definition of done
- [ ] Tests ensure that the behavior is correct
- [ ] The documentation is updated to reflect the changes done in this PR

### PR DESCRIPTION
**Pull request by semanticer**
### Summary
This PR should implement everything in #254. 

### Technical considerations
I still need to fix some tests but the main issue I want to discuss is that we don't have access to `cursorPosition` in our reducer. See my code comments below. 

### Why is it done like this and not some other way?
Please let me know if you have any ideas

### Relationships
Closes #254

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [ ] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests

### Pointers
Make sure that those tests are actually testing the right behavior because I'm not 100% sure.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Anyone

### CODE REVIEW

File: timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt [hunk 1]
```diff
@@ -256,6 +258,26 @@ class StartEditReducer @Inject constructor(
...
+                editableProject = EditableProject(
+                    name = project.name,
+                    color = project.color,
+                    active = project.active,
+                    isPrivate = project.isPrivate,
+                    billable = project.billable,
+                    workspaceId = project.workspaceId,
+                    clientId = project.clientId
+                ),
+                description = it.description.clearLastSubstringFromAnyTokenToPosition(charArrayOf('@')),
```
Comment (semanticer) [inline timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt:275]: `charArrayOf('@')` should be probably referenced from somewhere but the bigger issue here is that I'm using the default `cursorPosition` but I should be using the same `cursorPosition` that was used in `UpdateAutocompleteSuggestionsEffect`. Should we persist `cursorPosition` in our state?

Comment (daividssilverio) [reply timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt:275]: That's what I did here https://github.com/toggl/android/pull/303/commits/4dc1e4f0dadc03b0bee9c85e18e14dca85dd7181

File: timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt [hunk 2]
```diff
@@ -256,6 +258,26 @@ class StartEditReducer @Inject constructor(
             it.copy(editableProject = EditableProject(name = autocompleteSuggestion.name, workspaceId = 1))
         }
 
+    private fun StartEditState.modifyWithTaskSuggestion(autocompleteSuggestion: AutocompleteSuggestion.Task): StartEditState =
+        StartEditState.editableTimeEntry.modify(this) {
+            val taskSuggestion = autocompleteSuggestion.task
+            val project = this.projects[taskSuggestion.projectId] ?: throw ProjectDoesNotExistException()
+            it.copy(
+                editableProject = EditableProject(
```
Comment (daividssilverio) [inline timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt:266]: I was under the impression that the `EditableProject` was something used for new projects only.
Shouldn't we simply set the `projectId` here? If that's not the case, #303 is wrong. I actually set the `editableProject` to null when a project suggestion was picked.

Comment (semanticer) [reply timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt:266]: Ouu... I think you're right

Comment (daividssilverio) [reply timer/src/main/java/com/toggl/timer/startedit/domain/StartEditReducer.kt]: That was the idea, right @pdrj and @heytherewill (you've created the `EditableProject`).

**Pull request merged**